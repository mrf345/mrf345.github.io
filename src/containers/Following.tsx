import React from 'react'
import { connect } from 'react-redux'
import { Row } from 'react-bootstrap'
import { AnimationString } from 'react-animated-css'

import Panel from '../components/Panel'


interface ProjectsProps {
    following:Info[]
    loading:Loading
    animation:AnimationString
}


class Following extends React.Component<ProjectsProps> {
    componentDidMount() {
        window.scroll(0, 0)
    }

    render() {
        let counter = 0

        return this.props.following
                  .chunk(3)
                  .map((chunk, index) => {
                   return (<Row className="d-flex justify-content-center" key={Math.random().toString()}>
                       {chunk.map(user => <Panel className="text-center mt-3 mb-3 m-2"
                                                 goto={user.html_url}
                                                 key={user.id.toString()}
                                                 img={{src: user.avatar_url}}
                                                 animation={{animationIn: this.props.animation,
                                                             animationOut: 'fadeOutRight',
                                                             animationInDelay: ++counter * 100,
                                                             isVisible: true}}
                                                 clickAnimation={{animation: 'bounceOut'}}
                                                 header={{text:<i>({user.login})</i>,
                                                          className: 'h4 mt-2'}}
                                                 body={{text: user.html_url,
                                                        style: {fontSize: '120%'},
                                                        className: 'mt-1 mb-1'}} />)}
                   </Row>)})
    }
}


export default connect((state:State) => ({following: state.following,
                                          loading: state.loading}))(Following)
