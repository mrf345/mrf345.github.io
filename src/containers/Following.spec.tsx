import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import configureStore from 'redux-mock-store'

import Following from './Following'


describe("Testing Following container", () => {
    const mockStore = configureStore([])
    const user = {
        html_url: 'https://testing.com',
        id: 123123,
        avatar_url: 'https://testing.com/1.png',
        login: 'testing login'
    }
    let store:any

    beforeEach(() => {
        store = mockStore({following: Array(5).fill(user),
                           loading: {following: false}})
    })

    test("Sanity check Following contents", () => {
        const component = render(
            <Provider store={store}>
                <Following animation="fadeInDown" />
            </Provider>
        )

        expect(component).toMatchSnapshot()
    })
})
