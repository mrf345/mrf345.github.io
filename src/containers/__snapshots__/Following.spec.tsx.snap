// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Following container Sanity check Following contents 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-center row"
      >
        <div
          class="text-center mt-3 mb-3 m-2 col"
        >
          <div
            class="animated fadeInDown"
            style="animation-delay: 100ms; animation-duration: 1000ms; pointer-events: all;"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="h4 mt-2 col-12"
                >
                  <i>
                    (
                    testing login
                    )
                  </i>
                </div>
              </div>
              <div
                class="card-body"
              >
                <div
                  class="mb-2 col-12"
                >
                  <img
                    class="img-fluid rounded"
                    src="https://testing.com/1.png"
                  />
                </div>
                <div
                  class="mt-1 mb-1 col-12"
                  style="font-size: 120%;"
                >
                  https://testing.com
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="text-center mt-3 mb-3 m-2 col"
        >
          <div
            class="animated fadeInDown"
            style="animation-delay: 200ms; animation-duration: 1000ms; pointer-events: all;"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="h4 mt-2 col-12"
                >
                  <i>
                    (
                    testing login
                    )
                  </i>
                </div>
              </div>
              <div
                class="card-body"
              >
                <div
                  class="mb-2 col-12"
                >
                  <img
                    class="img-fluid rounded"
                    src="https://testing.com/1.png"
                  />
                </div>
                <div
                  class="mt-1 mb-1 col-12"
                  style="font-size: 120%;"
                >
                  https://testing.com
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="text-center mt-3 mb-3 m-2 col"
        >
          <div
            class="animated fadeInDown"
            style="animation-delay: 300ms; animation-duration: 1000ms; pointer-events: all;"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="h4 mt-2 col-12"
                >
                  <i>
                    (
                    testing login
                    )
                  </i>
                </div>
              </div>
              <div
                class="card-body"
              >
                <div
                  class="mb-2 col-12"
                >
                  <img
                    class="img-fluid rounded"
                    src="https://testing.com/1.png"
                  />
                </div>
                <div
                  class="mt-1 mb-1 col-12"
                  style="font-size: 120%;"
                >
                  https://testing.com
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-flex justify-content-center row"
      >
        <div
          class="text-center mt-3 mb-3 m-2 col"
        >
          <div
            class="animated fadeInDown"
            style="animation-delay: 400ms; animation-duration: 1000ms; pointer-events: all;"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="h4 mt-2 col-12"
                >
                  <i>
                    (
                    testing login
                    )
                  </i>
                </div>
              </div>
              <div
                class="card-body"
              >
                <div
                  class="mb-2 col-12"
                >
                  <img
                    class="img-fluid rounded"
                    src="https://testing.com/1.png"
                  />
                </div>
                <div
                  class="mt-1 mb-1 col-12"
                  style="font-size: 120%;"
                >
                  https://testing.com
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="text-center mt-3 mb-3 m-2 col"
        >
          <div
            class="animated fadeInDown"
            style="animation-delay: 500ms; animation-duration: 1000ms; pointer-events: all;"
          >
            <div
              class="card"
            >
              <div
                class="card-header"
              >
                <div
                  class="h4 mt-2 col-12"
                >
                  <i>
                    (
                    testing login
                    )
                  </i>
                </div>
              </div>
              <div
                class="card-body"
              >
                <div
                  class="mb-2 col-12"
                >
                  <img
                    class="img-fluid rounded"
                    src="https://testing.com/1.png"
                  />
                </div>
                <div
                  class="mt-1 mb-1 col-12"
                  style="font-size: 120%;"
                >
                  https://testing.com
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-center row"
    >
      <div
        class="text-center mt-3 mb-3 m-2 col"
      >
        <div
          class="animated fadeInDown"
          style="animation-delay: 100ms; animation-duration: 1000ms; pointer-events: all;"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <div
                class="h4 mt-2 col-12"
              >
                <i>
                  (
                  testing login
                  )
                </i>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="mb-2 col-12"
              >
                <img
                  class="img-fluid rounded"
                  src="https://testing.com/1.png"
                />
              </div>
              <div
                class="mt-1 mb-1 col-12"
                style="font-size: 120%;"
              >
                https://testing.com
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-center mt-3 mb-3 m-2 col"
      >
        <div
          class="animated fadeInDown"
          style="animation-delay: 200ms; animation-duration: 1000ms; pointer-events: all;"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <div
                class="h4 mt-2 col-12"
              >
                <i>
                  (
                  testing login
                  )
                </i>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="mb-2 col-12"
              >
                <img
                  class="img-fluid rounded"
                  src="https://testing.com/1.png"
                />
              </div>
              <div
                class="mt-1 mb-1 col-12"
                style="font-size: 120%;"
              >
                https://testing.com
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-center mt-3 mb-3 m-2 col"
      >
        <div
          class="animated fadeInDown"
          style="animation-delay: 300ms; animation-duration: 1000ms; pointer-events: all;"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <div
                class="h4 mt-2 col-12"
              >
                <i>
                  (
                  testing login
                  )
                </i>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="mb-2 col-12"
              >
                <img
                  class="img-fluid rounded"
                  src="https://testing.com/1.png"
                />
              </div>
              <div
                class="mt-1 mb-1 col-12"
                style="font-size: 120%;"
              >
                https://testing.com
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center row"
    >
      <div
        class="text-center mt-3 mb-3 m-2 col"
      >
        <div
          class="animated fadeInDown"
          style="animation-delay: 400ms; animation-duration: 1000ms; pointer-events: all;"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <div
                class="h4 mt-2 col-12"
              >
                <i>
                  (
                  testing login
                  )
                </i>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="mb-2 col-12"
              >
                <img
                  class="img-fluid rounded"
                  src="https://testing.com/1.png"
                />
              </div>
              <div
                class="mt-1 mb-1 col-12"
                style="font-size: 120%;"
              >
                https://testing.com
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-center mt-3 mb-3 m-2 col"
      >
        <div
          class="animated fadeInDown"
          style="animation-delay: 500ms; animation-duration: 1000ms; pointer-events: all;"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <div
                class="h4 mt-2 col-12"
              >
                <i>
                  (
                  testing login
                  )
                </i>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="mb-2 col-12"
              >
                <img
                  class="img-fluid rounded"
                  src="https://testing.com/1.png"
                />
              </div>
              <div
                class="mt-1 mb-1 col-12"
                style="font-size: 120%;"
              >
                https://testing.com
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
